src/crimson/osd/pg_backend.cc

```diff
@@ -209,13 +209,19 @@ PGBackend::read(const ObjectState& os, OSDOp& osd_op)
   return _read(oi.soid, offset, length, op.flags).safe_then(
     [&oi, &osd_op](auto&& bl) -> read_errorator::future<> {
     if (!_read_verify_data(oi, bl)) {
-      return crimson::ct_error::object_corrupted::make();
+      // crc mismatches
+      return crimson::ct_error::eagain::make();
```

Comment:
No.  Eventually, the error will become once you've triggered the repai, but at this point all that has happened is that read has concluded that the object is corrupted, and object_corrupted is a much more accurate description of the result than eagain.

Reply:
It looks like `load_metadata()` in `pg_backend` also returns `object_corrupted` https://github.com/ceph/ceph/blob/0aba110696b956581f7777147da27db589343f33/src/crimson/osd/pg_backend.cc#L88
Do we want to handle it the same way we handle the error returned from `read()` and `sparse_read()`? If not, what about returning a different dedicated error from the read methods?

Reply:
It probably would be valid to treat load_metadata errors the same way.  If not, you could introduce a metadata_corrupted error.